Source/Diagnostics/Diagnostics.cpp

```diff
@@ -103,6 +103,57 @@ Diagnostics::BaseReadParameters ()
 
     bool species_specified = pp.queryarr("species", m_species_names);
 
+    // Prepare to dump rho per species
```

Comment:
I feel this is a functional block (maybe even a good candidate for a member function). So we could potentially describe what it does from the beginning like this?
```suggestion
    // Prepare to dump rho per species:
    //   add rho_<species> to m_varnames if selected by the user
```